### PR TITLE
Security: DO-3367 set minimum version for certifi

### PR DESCRIPTION
**Pull request by patricia-causalens**
## Motivation and Context

Certifi library has a CVE against it: https://github.com/advisories/GHSA-248v-346w-9cwc

Added a minimum version for it, set to the patched version number as per the report above - `2024.7.4` 

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [ ] I have added relevant tests (unit, integration or regression).
- [ ] I have added comments to all the bits that are hard to follow.
- [ ] I have added/updated Documentation.
- [ ] I have updated the appropriate changelog with a line for my changes.

### CODE REVIEW

**Review by zack-causalens — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ anyio = ">=4.0.0"
 async-asgi-testclient = ">=1.4.11,<2.0.0"
 bokeh = ">=3.1.0,<3.2.0"
 cai-causal-graph = ">=0.3.6"
+certifi = ">=2024.7.4"
```
Comment (krzysztof-causalens) [inline pyproject.toml:15]: That's just the top-level pyproject which will be regenerated by anthology. The minimum version should be set in the packages themselves.

**Review by krzysztof-causalens — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (patricia-causalens) [reply pyproject.toml]: Oops, good catch fixed it

File: packages/dara-core/pyproject.toml [hunk 2]
```diff
@@ -50,7 +51,7 @@ openpyxl = "*"
 packaging = "^23.1"
 pandas = ">=1.1.0, <3.0.0"
 prometheus-client = "^0.14.1"
-pyarrow = "*"
+pyarrow = "<17.0.0"
```
Comment (krzysztof-causalens) [inline packages/dara-core/pyproject.toml:54]: What's the reason to restrict pyarrow here?

Comment (patricia-causalens) [reply packages/dara-core/pyproject.toml]: Was just testing, the version had just been released and was not being picked up yet so was just checking the issue was not elsewhere. Seems to be picking it up now though so have removed it

**Review by krzysztof-causalens — APPROVED** [review]

**Pull request merged**